crates/aptos-api-tester/src/tests/coin_transfer.rs

```diff
@@ -0,0 +1,273 @@
+use aptos_rest_client::Client;
+use aptos_sdk::{coin_client::CoinClient, types::LocalAccount};
+use aptos_types::account_address::AccountAddress;
+
+const TRANSFER_AMOUNT: u64 = 1_000;
+
+/// Tests coin transfer. Checks that:
+///   - receiver balance reflects transferred amount
+///   - receiver balance shows correct amount at the previous version
+pub async fn test(network_name: NetworkName, run_id: &str) -> Result<(), TestFailure> {
+    // setup
+    let (client, mut account, receiver) = emit_step_metrics(
+        time_fn!(setup, network_name),
+        TestName::CoinTransfer,
+        SETUP,
+        network_name,
+        run_id,
+    )?;
+    let coin_client = CoinClient::new(&client);
+
+    // check account data persistently
+    emit_step_metrics(
+        time_fn!(
+            persistent_check::address_address,
+            CHECK_ACCOUNT_DATA,
+            check_account_data,
+            &client,
+            account.address(),
+            receiver
+        ),
+        TestName::CoinTransfer,
+        CHECK_ACCOUNT_DATA,
+        network_name,
+        run_id,
+    )?;
+
+    // transfer coins to the receiver
+    let version = emit_step_metrics(
+        time_fn!(
+            transfer_coins,
+            &client,
+            &coin_client,
+            &mut account,
+            receiver
+        ),
+        TestName::CoinTransfer,
+        TRANSFER_COINS,
+        network_name,
+        run_id,
+    )?;
+
+    // check receiver balance persistently
+    emit_step_metrics(
+        time_fn!(
+            persistent_check::address,
+            CHECK_ACCOUNT_BALANCE,
+            check_account_balance,
+            &client,
+            receiver
+        ),
+        TestName::CoinTransfer,
+        CHECK_ACCOUNT_BALANCE,
+        network_name,
+        run_id,
+    )?;
+
+    // check receiver balance at previous version persistently
+    emit_step_metrics(
+        time_fn!(
+            persistent_check::address_version,
+            CHECK_ACCOUNT_BALANCE_AT_VERSION,
+            check_account_balance_at_version,
+            &client,
+            receiver,
+            version
+        ),
+        TestName::CoinTransfer,
+        CHECK_ACCOUNT_BALANCE_AT_VERSION,
+        network_name,
+        run_id,
+    )?;
+
+    Ok(())
+}
+
+// Steps
+
+async fn setup(
+    network_name: NetworkName,
+) -> Result<(Client, LocalAccount, AccountAddress), TestFailure> {
+    // spin up clients
+    let client = get_client(network_name);
+    let faucet_client = get_faucet_client(network_name);
+
+    // create account
+    let account = match create_and_fund_account(&faucet_client).await {
+        Ok(account) => account,
+        Err(e) => {
+            info!(
+                "test: coin_transfer part: setup ERROR: {}, with error {:?}",
+                ERROR_COULD_NOT_FUND_ACCOUNT, e
+            );
+            return Err(e.into());
+        },
+    };
+    info!(
+        "test: coin_transfer part: setup creating account: {}",
```

Comment:
This is actually the only message of its kind and serves the purpose of communicating the account address in case we want to manually check from the explorer, the success messages are returned in the `emit_step_latency` function

Reply:
Okay cool works for me.